src/Admin_Tools_Command.php

```diff
@@ -0,0 +1,301 @@
+<?php
+
+/**
+ * Enables/Disables admin-tools on a site.
+ *
+ * ## EXAMPLES
+ *
+ *     # Enable admin tools on site
+ *     $ ee admin-tools up example.com
+ *
+ * @package ee-cli
+ */
+
+use EE\Model\Site;
+use \Symfony\Component\Filesystem\Filesystem;
+use \Composer\Console\Application;
+use \Symfony\Component\Console\Input\ArrayInput;
+
+class Admin_Tools_Command extends EE_Command {
+
+
+	/**
+	 * @var string $command Name of the command being run.
+	 */
+	private $command;
+
+	/**
+	 * @var Filesystem $fs Symfony Filesystem object.
+	 */
+	private $fs;
+
+	/**
+	 * @var array $db Object containing essential site related information.
+	 */
+	private $db;
+
+	public function __construct() {
+
+		$this->command = 'admin-tools';
+		$this->fs      = new Filesystem();
+		define( 'ADMIN_TOOL_DIR', EE_CONF_ROOT . '/admin-tools' );
+	}
+
+	/**
+	 * Installs admin-tools for EasyEngine.
+	 */
+	public function install() {
+
+		if ( ! $this->is_installed() ) {
+			EE::log( 'Installing admin-tools. This may take some time.' );
+			$this->fs->mkdir( ADMIN_TOOL_DIR );
+		}
+
+		$tools = json_decode( file_get_contents( ADMIN_TOOLS_FILE ), true );
```

Comment:
a couple of things here we can take care of.
1. if we are expecting the ADMIN_TOOLS_FILE file to be .json file we can check the file extension.
2. check if `file_get_contents( ADMIN_TOOLS_FILE )` is not empty if it is do not proceed further just add an error message.
3. Check if json_decode is done correctly ( it will fail if there is invalid JSON formate data ) After this  line you can check `if ( json_last_error() != JSON_ERROR_NONE )` and send a error message/exception.
4. And one last thing if json data is valid but empty `{}` then you can add a check for `$tools` if it's empty add a message/exception.

Reply:
Will update according to this :+1: